src/java.base/share/classes/java/util/WeakHashMap.java

```diff
@@ -36,13 +39,35 @@
  * Hash table based implementation of the {@code Map} interface, with
  * <em>weak keys</em>.
  * An entry in a {@code WeakHashMap} will automatically be removed when
- * its key is no longer in ordinary use.  More precisely, the presence of a
+ * its key is no longer in ordinary use.
+ * More precisely, for keys that are identity objects, the presence of a
  * mapping for a given key will not prevent the key from being discarded by the
  * garbage collector, that is, made finalizable, finalized, and then reclaimed.
+ * For keys that are {@linkplain Class#isValue() Value objects}, the retention of the
```

Comment:
should this be "value objects" rather than "Value objects"?  This comment applies to all occurrences.